spring-messaging/src/main/java/org/springframework/messaging/simp/stomp/DefaultStompSession.java

```diff
@@ -328,12 +328,19 @@ public Receiptable acknowledge(String messageId, boolean consumed) {
 			stompHeaders.setId(messageId);
 		}
 
-		String receiptId = checkOrAddReceipt(stompHeaders);
+		Receiptable receiptable = acknowledge(stompHeaders, consumed);
```

Comment:
This looks a case for a straight `return acknowledge(stompHeaders, consumed)` call.

On a related note, @rstoyanchev, there seems to be inconsistent naming of `StompHeaders` arguments, primarily between `StompSession` and implementing methods in `DefaultStompSession`, but also within the `StompSession` interface itself regarding the new `unsubscribe(StompHeaders)` variant. Let's consistently name those arguments `headers`, replacing all current `stompHeaders` names.